File: algorithms/easy/count_the_number_of_vowel_strings_in_range.dart
class Solution {
  int vowelStrings(List<String> words, int left, int right) {
    RegExp exp = RegExp(r'^[aeiou](.*[aeiou])?$');
    int ct = 0;
    for (String w in words.sublist(left, right + 1)) {
      if (exp.hasMatch(w)) {
        ct += 1;
      }
    }
    return ct;
  }
}

File: algorithms/easy/count_the_number_of_vowel_strings_in_range.ex
defmodule Solution do
  @spec vowel_strings(words :: [String.t()], left :: integer, right :: integer) :: integer
  def vowel_strings(words, left, right) do
    words
    |> Enum.drop(left)
    |> Enum.take(right - left + 1)
    |> Enum.count(fn w ->
      Regex.match?(~r/^[aeiou](.*[aeiou])?$/, w)
    end)
  end
end
